### ISSUE TITLE
Field visibility elected at runtime should be reflected in exported CSV

### ISSUE THREAD

**Issue by avezina** [Esri/solutions-components#452]
If I choose to hide fields at runtime, this should affect which fields are exported in the CSV.

![image](https://github.com/Esri/solutions-components/assets/50338557/9e25279c-bc47-4a1a-8cdb-5d1bb2e21eb7)

Related: https://github.com/Esri/solutions-components/issues/350

**Comment by azizaparveen** [comment]
@jmhauck will ObjectID always show up in the download even if it is hidden in the table?
![image](https://github.com/Esri/solutions-components/assets/138827632/728c675d-e6d3-4498-b7fe-0be914a631cf)

**Comment by azizaparveen** [comment]
This issue is not reproducible in devext, only reproducible in qa

**Comment by jmhauck** [comment]
@azizaparveen oh sweet...maybe it will bubble up to QA. When I looked at the end of the week last week before heading out I could only repro in dev and not prod. But it was reproducible at the REST api level. 

Just for notes hopefully...Only thing that I noticed at that time was the OID would not be forced in dev if no resultOffset was provided but I wasn't seeing where we would have a chance deal with it somehow in our code.
